src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -0,0 +1,188 @@
+  private long desiredCacheMaxSizeInBytes;
+
+  private Map<ResourceEnum, Long> stepSizeInBytes = new HashMap<>();
+
+  public ModifyCacheMaxSizeAction(
+      final NodeKey esNode,
+      final ResourceEnum cacheType,
+      final NodeConfigCache nodeConfigCache,
+      final double cacheSizeUpperBound,
+      final boolean increase) {
+    // TODO: Add lower bound for caches
+    // TODO: Address cache scaling down  when JVM decider is available
+
+    this.esNode = esNode;
+    this.cacheType = cacheType;
+    this.nodeConfigCache = nodeConfigCache;
+    this.cacheSizeUpperBound = cacheSizeUpperBound;
+
+    setStepSize();
+
+    long desiredCapacity;
+    if (validateAndSetNodeConfigValues(esNode, cacheType)) {
+
+      setBounds();
+      desiredCapacity =
+          increase
+              ? currentCacheMaxSizeInBytes + getStepSize(cacheType)
+              : currentCacheMaxSizeInBytes;
+    } else {
+      desiredCapacity = currentCacheMaxSizeInBytes;
+    }
+
+    setDesiredCacheMaxSize(desiredCapacity);
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public boolean isActionable() {
+    return desiredCacheMaxSizeInBytes != 0
+        && desiredCacheMaxSizeInBytes != currentCacheMaxSizeInBytes;
+  }
+
+  @Override
+  public long coolOffPeriodInMillis() {
+    return COOL_OFF_PERIOD_IN_MILLIS;
+  }
+
+  @Override
+  public List<NodeKey> impactedNodes() {
+    return Collections.singletonList(esNode);
+  }
+
+  @Override
+  public Map<NodeKey, ImpactVector> impact() {
+    final ImpactVector impactVector = new ImpactVector();
+    if (desiredCacheMaxSizeInBytes > currentCacheMaxSizeInBytes) {
+      impactVector.increasesPressure(HEAP);
+    } else if (desiredCacheMaxSizeInBytes < currentCacheMaxSizeInBytes) {
+      impactVector.decreasesPressure(HEAP);
+    }
+    return Collections.singletonMap(esNode, impactVector);
+  }
+
+  @Override
+  public String summary() {
+    if (!isActionable()) {
+      return String.format("No action to take for: [%s]", NAME);
+    }
+    return String.format(
+        "Update [%s] capacity from [%d] to [%d] on node [%s]",
+        cacheType.toString(),
+        currentCacheMaxSizeInBytes,
+        desiredCacheMaxSizeInBytes,
+        esNode.getNodeId());
+  }
+
+  @Override
+  public String toString() {
+    return summary();
+  }
+
+  public void setStepSizeForCache(final ResourceEnum cacheType, final long stepSize) {
+    stepSizeInBytes.put(cacheType, stepSize);
+  }
+
+  public Long getCurrentCacheMaxSizeInBytes() {
+    return currentCacheMaxSizeInBytes;
+  }
+
+  public Long getDesiredCacheMaxSizeInBytes() {
+    return desiredCacheMaxSizeInBytes;
+  }
+
+  public ResourceEnum getCacheType() {
+    return cacheType;
+  }
+
+  private boolean validateAndSetNodeConfigValues(
+      final NodeKey esNode, final ResourceEnum cacheType) {
+    Long cacheMaxSize = NodeConfigReaderUtil.getCacheMaxSizeInBytes(nodeConfigCache, esNode, cacheType);
```

Comment:
should we decouple this reading from cache operation from the Action class and let decider to read and pass them in as inputs args? Any thought ?

Reply:
Had an offline discussion with vigya,

Reply:
Actually there is confusion with naming. We dont want to pass *upper/lower bounds* from deciders. It may be okay to pass currentCapacity if needed (similar to queues). However, since configs come from rca.conf which is read in `Node` objects, deciders have access to them.

We should create separate ActionConfig objects for these action configs. decider configs should hold decider related settings like enable/disable deciders or deciderPeriod. These action configs can then be passed to each action while initializing it. Also we can split the cache action into separate actions since their RCAs are already separate. (It's ok to do this in a separate PR).